reference/format/search.rst

```diff
@@ -1,2 +1,166 @@
+|           |   * :ref:`name <reference-format-search-fields-name>`      |
+|           |   * `relation_fields`_                                     |
+|           |   * `relation_type`_                                       |
+|           |   * `target_fields`_                                       |
+|           |   * `target_type`_                                         |
+|           |                                                            |
+|           | * `label`_                                                 |
+|           | * `mode`_                                                  |
+|           | * `route`_                                                 |
+|           |                                                            |
+|           |   * :ref:`name <reference-format-search-route-name>`       |
+|           |   * `parameters`_                                          |
+|           |                                                            |
+|           | * `search_template`_                                       |
+|           | * `title_fields`_                                          |
++-----------+------------------------------------------------------------+
+
+``alias``
+---------
+
+**type**: ``string``
+
+An alias used in the ``from`` keyword in :ref:`advanced search <advanced-search-api>`.
+
+.. _reference-format-search-fields:
+
+``fields``
+----------
+
+**type**: ``sequence``
+
+The list of fields that will be added to the search index. Each entry must be a map that can use
+the following keys to configure how the property value will be indexed:
+
+.. _reference-format-search-fields-name:
+
+``name``
+~~~~~~~~
+
+**type**: ``string``
+
+The name of the entity property. This option is required.
+
+``relation_fields``
+~~~~~~~~~~~~~~~~~~~
+
+**type**: ``sequence``
+
+When the field represents an association (i.e. a value is configured for `relation_type`_), this is
+a list of fields from the target entity to index. For each entry all the options of the parent
+:ref:`fields option <reference-format-search-fields>` apply.
+
+``relation_type``
+~~~~~~~~~~~~~~~~~
+
+**type**: ``string``
+
+When the property denotes an association with another entity, the type of association (one of
+``one-to-one``, ``many-to-many``, ``one-to-many``, or ``many-to-one``) must be configured with this
+option.
+
+``target_fields``
+~~~~~~~~~~~~~~~~~
+
+**type**: ``sequence``
+
+The ``target_fields`` option list the named indexes to which the property value will be added.
+
+For example, a contact may have the properties ``firstName``, ``lastName``, and ``namePrefix`` and
+all three properties should be searched when the user is loooking for a value in the virtual
+``name`` field (when using the advanced search API). In this case, all three properties will list
+the ``name`` field in ``target_fields``:
+
+.. code-block:: yaml
+    :linenos:
+
+    search:
+        fields:
```

Comment:
Now you describes the `target_fields ` parameter. But this is an example of `fields` block.